cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsMessageCallback.java

```diff
@@ -89,16 +91,16 @@ public void onReceived(Iterator<FetchedMessage> messages) {
       addProcessingStats(records);
       metricStore.add(records);
     } catch (Exception e) {
-      String msg = "Failed to add metrics data to a store";
-      LOG.error(msg);
-      // todo: will it shut down the whole the metrics processor service??
-      throw new RuntimeException(msg, e);
+      // SimpleKafkaConsumer will log the error, and continue on past these messages
+      throw new RuntimeException("Failed to add metrics data to a store", e);
     }
 
-    recordProcessed += records.size();
-    if (recordProcessed % 1000 == 0) {
-      LOG.info("{} metrics records processed", recordProcessed);
-      LOG.info("Last record time: {}", records.get(records.size() - 1).getTimestamp());
+    recordsProcessed += records.size();
+    // avoid logging more than once a minute
+    if (System.currentTimeMillis() > lastLoggedMillis + TimeUnit.MINUTES.toMillis(1)) {
+      lastLoggedMillis = System.currentTimeMillis();
+      LOG.info("{} metrics records processed. Last record time: {}.",
```

Comment:
To reduce verbosity, you could combine these tow messages as they are always logged together. 
Also, I would rephrase to: N metrics records processed in the last minute.